Assets/Scripts/TextVolume.cs

```diff
@@ -0,0 +1,56 @@
+﻿using UnityEngine;
+using System.Collections;
+
+public class TextVolume : MonoBehaviour
+{
+	[SerializeField] string _text = "";
+	public string text
+	{
+		get { return _text; }
+	}
+
+	[SerializeField] bool _fadeOut = true;
+	[SerializeField] float _timeUntilFadeout = 7.0f;
+	[SerializeField] float _fadeoutDuration = 2.0f;
+
+	[SerializeField] bool _triggerOnce = true;
+
+	[ReadOnly]
+	[SerializeField] bool _hasBeenTriggered = false;
+
+	private TextBox _textBox;
+
+	void Awake()
+	{
+		_textBox = FindObjectOfType<TextBox>();
+	}
+
+	void OnDrawGizmos()
+	{
+		Gizmos.color = Color.blue;
+		Transform transform = GetComponent<Transform>();
+		Gizmos.matrix = Matrix4x4.TRS( transform.position, transform.rotation, transform.lossyScale );
```

Comment:
I think there's a shorthand for this, something like transform.localToWorldMatrix